src/einsteinpy/geodesic.py

```diff
@@ -157,28 +220,28 @@ def calculate_trajectory_iterator(
         ODE = RK45(
             fun=self.metric.f_vec,
             t0=0.0,
-            y0=self.init_vec,
+            y0=self.state,
             t_bound=1e300,
             **OdeMethodKwargs,
         )
 
-        _scr = self.metric.sch_rad * 1.001
+        g = self.metric
 
-        while True:
-            if not return_cartesian:
-                yield ODE.t, ODE.y
+        _scr = g.sch_rad * 1.001
 
-            else:
-                conv_coords = {
-                    "S": SphericalConversion,
-                    "BL": BoyerLindquistConversion,
-                }
+        while True:
+            if return_cartesian:
+                # Getting the corresponding Conversion class
+                type_coords_conv = type(self.coords).__mro__[1]
```

Comment:
Not a very clean code

Reply:
Based on the conversation in the chat room:
```
coords = self.coords
# Proxy object of the Conversion superclass
conv = super(type(coords), coords)
cart_vecs = list()
for v in vecs:
    conv.__init__(
        v[0], v[1], v[2], v[3], v[5], v[6], v[7]
    )
    vals = conv.convert_cartesian(M=g.M, a=g.a)
    cart_vecs.append(np.hstack((vals[:4], v[4], vals[4:])))
```
This method is object-oriented and coordinate-agnostic. And just a tad slower than `mro` or `bases`.

Reply:
I feel its doable. We just need to put one/two extra lines of comment explaining the process.

Reply:
Moving back to previous code, using `bases`. Both have similar performance.